src/server/game/Spells/SpellEffects.cpp

```diff
@@ -1742,6 +1742,86 @@ void Spell::EffectEnergize(SpellEffIndex effIndex)
     if (!unitTarget->IsAlive())
         return;
 
+    // Handle Secondary Potion Effects here
+    // Mad Alchemist's Potion (34440)
+    if (m_spellInfo->Id == 45051)
+    {
+        std::set<uint32> availableElixirs;
```

Comment:
Would you mind telling me wich spell script function to use? I dont know much about spell scripts yet :/